### ISSUE TITLE
Docker Build - Failing

### ISSUE THREAD

**Issue by shreyas3336** [crytic/echidna#689]
```
>docker build echidna
[+] Building 2.0s (12/21)
 => [internal] load build definition from Dockerfile                                                                                                                                                          0.1s
 => => transferring dockerfile: 1.64kB                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.1s
 => => transferring context: 34B                                                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/library/ubuntu:bionic                                                                                                                                              1.4s
 => [internal] load build context                                                                                                                                                                             0.1s
 => => transferring context: 14.21kB                                                                                                                                                                          0.0s
 => [builder 1/8] FROM docker.io/library/ubuntu:bionic@sha256:9bc830af2bef73276515a29aa896eedfa7bdf4bdbc5c1063b4c457a4bbb8cd79                                                                                0.0s
 => CACHED [final  2/10] WORKDIR /root                                                                                                                                                                        0.0s
 => CACHED [builder 2/8] RUN apt-get update && apt-get -y upgrade && apt-get install -y sudo git cmake curl libgmp-dev libssl-dev libbz2-dev libreadline-dev software-properties-common libsecp256k1-dev &&   0.0s
 => CACHED [builder 3/8] RUN sudo apt-get install -y build-essential                                                                                                                                          0.0s
 => CACHED [builder 4/8] RUN wget -qO- https://get.haskellstack.org/ | sh                                                                                                                                     0.0s
 => CACHED [builder 5/8] COPY . /echidna/                                                                                                                                                                     0.0s
 => CACHED [builder 6/8] WORKDIR /echidna                                                                                                                                                                     0.0s
 => ERROR [builder 7/8] RUN .github/scripts/install-libff.sh                                                                                                                                                  0.4s
------
 > [builder 7/8] RUN .github/scripts/install-libff.sh:
#12 0.319 /bin/sh: 1: .github/scripts/install-libff.sh: not found
------
executor failed running [/bin/sh -c .github/scripts/install-libff.sh]: exit code: 127
```

**Comment by elopez** [comment]
Hi @shreyas3336, I just tried building the repo `Dockerfile` on the `master` branch and I was not able to reproduce the issue. Can you provide more details on the issue you are experiencing?

```shell
~$ git clone https://github.com/crytic/echidna
Cloning into 'echidna'...
remote: Enumerating objects: 8619, done.
remote: Counting objects: 100% (953/953), done.
remote: Compressing objects: 100% (494/494), done.
remote: Total 8619 (delta 525), reused 821 (delta 434), pack-reused 7666
Receiving objects: 100% (8619/8619), 1.65 MiB | 12.67 MiB/s, done.
Resolving deltas: 100% (5132/5132), done.
~$ cd echidna/
~/echidna$ ls
CHANGELOG.md  CONTRIBUTING.md  Dockerfile  LICENSE  README.md  action  default.nix  echidna.png  examples  lib  package.yaml  src  stack.yaml
~/echidna$ docker build .
Sending build context to Docker daemon  951.8kB
Step 1/21 : FROM ubuntu:bionic AS builder
bionic: Pulling from library/ubuntu
284055322776: Already exists 
Digest: sha256:0fedbd5bd9fb72089c7bbca476949e10593cebed9b1fb9edf5b79dbbacddd7d6
Status: Downloaded newer image for ubuntu:bionic
 ---> 5a214d77f5d7
Step 2/21 : ENV LD_LIBRARY_PATH=/usr/local/lib PREFIX=/usr/local HOST_OS=Linux
 ---> Running in 419aa34bb8af
Removing intermediate container 419aa34bb8af
 ---> 27157f58df8e
Step 3/21 : RUN apt-get update && apt-get -y upgrade && apt-get install -y sudo git cmake curl libgmp-dev libssl-dev libbz2-dev libreadline-dev software-properties-common libsecp256k1-dev && apt-get clean
 ---> Running in 5adcbd1e51ef
Get:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic InRelease [242 kB]
Get:3 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [2461 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:5 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [26.8 kB]
Get:6 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [691 kB]
Get:7 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1446 kB]
Get:8 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Get:9 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [11.3 MB]
Get:10 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages [186 kB]
Get:11 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1344 kB]
Get:12 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages [13.5 kB]
Get:13 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [2898 kB]
Get:14 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [725 kB]
Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [34.4 kB]
Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2227 kB]
Get:17 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages [11.4 kB]
Get:18 http://archive.ubuntu.com/ubuntu bionic-backports/main amd64 Packages [11.3 kB]
Fetched 23.9 MB in 4s (5646 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  binutils binutils-common binutils-x86-64-linux-gnu bzip2-doc ca-certificates
(...)
Setting up cmake (3.10.2-1ubuntu2.18.04.2) ...
Setting up git (1:2.17.1-1ubuntu0.9) ...
Setting up curl (7.58.0-2ubuntu3.16) ...
Setting up gnupg (2.2.4-1ubuntu1.4) ...
Processing triggers for libc-bin (2.27-3ubuntu1.4) ...
Processing triggers for ca-certificates (20210119~18.04.2) ...
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
Processing triggers for dbus (1.12.2-1ubuntu1.2) ...
Removing intermediate container 5adcbd1e51ef
 ---> dd2a1cbdd2ef
Step 4/21 : RUN curl -sSL https://get.haskellstack.org/ | sh
 ---> Running in 5495522c2128
Detected Linux distribution: ubuntu

Installing dependencies...


About to use 'sudo' to run the following command as root:
    apt-get install -y g++ libffi-dev zlib1g-dev
in order to install required system dependencies.

Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  g++-7 libstdc++-7-dev
Suggested packages:
  g++-multilib g++-7-multilib gcc-7-doc libstdc++6-7-dbg libstdc++-7-doc
The following NEW packages will be installed:
  g++ g++-7 libffi-dev libstdc++-7-dev zlib1g-dev
0 upgraded, 5 newly installed, 0 to remove and 0 not upgraded.
Need to get 11.5 MB of archives.
After this operation, 43.2 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libstdc++-7-dev amd64 7.5.0-3ubuntu1~18.04 [1471 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 g++-7 amd64 7.5.0-3ubuntu1~18.04 [9697 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 g++ amd64 4:7.4.0-1ubuntu2.3 [1568 B]
Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 zlib1g-dev amd64 1:1.2.11.dfsg-0ubuntu2 [176 kB]
Get:5 http://archive.ubuntu.com/ubuntu bionic/main amd64 libffi-dev amd64 3.2.1-8 [156 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 11.5 MB in 2s (5816 kB/s)
Selecting previously unselected package libstdc++-7-dev:amd64.
(Reading database ... 17784 files and directories currently installed.)
Preparing to unpack .../libstdc++-7-dev_7.5.0-3ubuntu1~18.04_amd64.deb ...
Unpacking libstdc++-7-dev:amd64 (7.5.0-3ubuntu1~18.04) ...
Selecting previously unselected package g++-7.
Preparing to unpack .../g++-7_7.5.0-3ubuntu1~18.04_amd64.deb ...
Unpacking g++-7 (7.5.0-3ubuntu1~18.04) ...
Selecting previously unselected package g++.
Preparing to unpack .../g++_4%3a7.4.0-1ubuntu2.3_amd64.deb ...
Unpacking g++ (4:7.4.0-1ubuntu2.3) ...
Selecting previously unselected package zlib1g-dev:amd64.
Preparing to unpack .../zlib1g-dev_1%3a1.2.11.dfsg-0ubuntu2_amd64.deb ...
Unpacking zlib1g-dev:amd64 (1:1.2.11.dfsg-0ubuntu2) ...
Selecting previously unselected package libffi-dev:amd64.
Preparing to unpack .../libffi-dev_3.2.1-8_amd64.deb ...
Unpacking libffi-dev:amd64 (3.2.1-8) ...
Setting up libffi-dev:amd64 (3.2.1-8) ...
Setting up libstdc++-7-dev:amd64 (7.5.0-3ubuntu1~18.04) ...
Setting up zlib1g-dev:amd64 (1:1.2.11.dfsg-0ubuntu2) ...
Setting up g++-7 (7.5.0-3ubuntu1~18.04) ...
Setting up g++ (4:7.4.0-1ubuntu2.3) ...
update-alternatives: using /usr/bin/g++ to provide /usr/bin/c++ (c++) in auto mode
update-alternatives: warning: skip creation of /usr/share/man/man1/c++.1.gz because associated file /usr/share/man/man1/g++.1.gz (of link group c++) doesn't exist

Using generic bindist...

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   138  100   138    0     0    597      0 --:--:-- --:--:-- --:--:--   597
100   637  100   637    0     0   2068      0 --:--:-- --:--:-- --:--:--  2068
100 17.1M  100 17.1M    0     0  26.9M      0 --:--:-- --:--:-- --:--:-- 26.9M
Installing Stack to: /usr/local/bin/stack...

-------------------------------------------------------------------------------

Stack has been installed to: /usr/local/bin/stack

WARNING: '/root/.local/bin' is not on your PATH.
    Stack will place the binaries it builds in '/root/.local/bin' so
    for best results, please add it to the beginning of PATH in your profile.

Removing intermediate container 5495522c2128
 ---> a1906bd628ee
Step 5/21 : COPY . /echidna/
 ---> f353b82fef96
Step 6/21 : WORKDIR /echidna
 ---> Running in 259a54bdfc0a
Removing intermediate container 259a54bdfc0a
 ---> 65dc51cbbb4d
Step 7/21 : RUN .github/scripts/install-libff.sh
 ---> Running in c08947e69b24
+ source .github/scripts/host.sh
++ '[' -z x ']'
++ '[' Linux = Linux ']'
++ EXT=.a
+ '[' -f /root/.local/lib/libff.a ']'
+ git clone https://github.com/scipr-lab/libff --recursive
Cloning into 'libff'...
Submodule 'depends/ate-pairing' (https://github.com/herumi/ate-pairing.git) registered for path 'depends/ate-pairing'
Submodule 'gtest' (https://github.com/google/googletest.git) registered for path 'depends/gtest'
Submodule 'depends/xbyak' (https://github.com/herumi/xbyak.git) registered for path 'depends/xbyak'
Cloning into '/echidna/libff/depends/ate-pairing'...
Cloning into '/echidna/libff/depends/gtest'...
Cloning into '/echidna/libff/depends/xbyak'...
Submodule path 'depends/ate-pairing': checked out 'e69890125746cdaf25b5b51227d96678f76479fe'
Submodule path 'depends/gtest': checked out 'a325ad2db5deb623eab740527e559b81c0f39d65'
Submodule path 'depends/xbyak': checked out 'f0a8f7faa27121f28186c2a7f4222a9fc66c283d'
+ cd libff
+ git submodule init
+ git submodule update
+ git checkout v0.2.1
warning: unable to rmdir 'depends/gtest': Directory not empty
Note: checking out 'v0.2.1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 5835b8c Merge branch 'clearmatics-fix-curves-interfaces-inconsistencies'
+ ARGS='-DCMAKE_INSTALL_PREFIX=/usr/local -DWITH_PROCPS=OFF'
+ CXXFLAGS=
+ '[' Linux = macOS ']'
+ mkdir build
+ cd build
+ CXXFLAGS='-fPIC '
+ cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DWITH_PROCPS=OFF ..
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- GMP: /usr/lib/x86_64-linux-gnu/libgmp.so, /usr/include/x86_64-linux-gnu
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.1") 
-- Configuring done
-- Generating done
-- Build files have been written to: /echidna/libff/build
+ make
Scanning dependencies of target zm
[  2%] Building CXX object depends/CMakeFiles/zm.dir/ate-pairing/src/zm.cpp.o
[  5%] Building CXX object depends/CMakeFiles/zm.dir/ate-pairing/src/zm2.cpp.o
[  8%] Linking CXX static library libzm.a
[  8%] Built target zm
Scanning dependencies of target ff
[ 11%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_g1.cpp.o
[ 14%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_g2.cpp.o
[ 17%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_init.cpp.o
[ 20%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_pairing.cpp.o
[ 23%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_pp.cpp.o
[ 26%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_g1.cpp.o
[ 29%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_g2.cpp.o
[ 32%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_init.cpp.o
[ 35%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_pairing.cpp.o
[ 38%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_pp.cpp.o
[ 41%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_g1.cpp.o
[ 44%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_g2.cpp.o
[ 47%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_init.cpp.o
[ 50%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_pairing.cpp.o
[ 52%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_pp.cpp.o
[ 55%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt46_common.cpp.o
[ 58%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_g1.cpp.o
[ 61%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_g2.cpp.o
[ 64%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_init.cpp.o
[ 67%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_pairing.cpp.o
[ 70%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_pp.cpp.o
[ 73%] Building CXX object libff/CMakeFiles/ff.dir/common/double.cpp.o
[ 76%] Building CXX object libff/CMakeFiles/ff.dir/common/profiling.cpp.o
/echidna/libff/libff/common/profiling.cpp: In function 'void libff::print_mem(const string&)':
/echidna/libff/libff/common/profiling.cpp:333:35: warning: unused parameter 's' [-Wunused-parameter]
 void print_mem(const std::string &s)
                                   ^
[ 79%] Building CXX object libff/CMakeFiles/ff.dir/common/utils.cpp.o
[ 82%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_g1.cpp.o
[ 85%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_g2.cpp.o
[ 88%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_gt.cpp.o
[ 91%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_init.cpp.o
[ 94%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_pairing.cpp.o
[ 97%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_pp.cpp.o
[100%] Linking CXX static library libff.a
[100%] Built target ff
+ make install
[  8%] Built target zm
[100%] Built target ff
Install the project...
-- Install configuration: "RelWithDebInfo"
-- Installing: /usr/local/include/libff
-- Installing: /usr/local/include/libff/algebra
-- Installing: /usr/local/include/libff/algebra/fields
-- Installing: /usr/local/include/libff/algebra/fields/fp6_2over3.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp_aux.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp3.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp6_2over3.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp4.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp3.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp12_2over3over2.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp.tcc
-- Installing: /usr/local/include/libff/algebra/fields/field_utils.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp2.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp12_2over3over2.tcc
-- Installing: /usr/local/include/libff/algebra/fields/bigint.tcc
-- Installing: /usr/local/include/libff/algebra/fields/field_utils.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp4.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp2.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp6_3over2.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp.hpp
(...)
```

**Comment by elopez** [comment]
I'm closing this issue now, but feel free to reopen it if you encounter this again!

**Comment by FIVIL** [comment]
Hi, I was trying to build echidna but I am running into the same problem, I tried building it multiple times and cloned the repo several times, but no luck.  

PS C:\Users\hamme\Desktop\Smart Contract\slither> git clone https://github.com/crytic/echidna.git
Cloning into 'echidna'...
remote: Enumerating objects: 9057, done.
remote: Counting objects: 100% (1391/1391), done.
remote: Compressing objects: 100% (652/652), done.
remote: Total 9057 (delta 825), reused 1228 (delta 704), pack-reused 7666
Receiving objects: 100% (9057/9057), 1.70 MiB | 8.63 MiB/s, done.
Resolving deltas: 100% (5432/5432), done.
PS C:\Users\hamme\Desktop\Smart Contract\slither> cd .\echidna\


    Directory: C:\Users\hamme\Desktop\Smart Contract\slither\echidna


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         5/18/2022   5:25 PM                .github
d-----         5/18/2022   5:25 PM                lib
d-----         5/18/2022   5:25 PM                nix
d-----         5/18/2022   5:25 PM                src
d-----         5/18/2022   5:25 PM                tests
-a----         5/18/2022   5:25 PM            124 .codeclimate.yml
-a----         5/18/2022   5:25 PM             26 .dockerignore
-a----         5/18/2022   5:25 PM             34 .gitattributes
-a----         5/18/2022   5:25 PM            248 .gitignore
-a----         5/18/2022   5:25 PM           8131 CHANGELOG.md
-a----         5/18/2022   5:25 PM           2238 CONTRIBUTING.md
-a----         5/18/2022   5:25 PM           4153 default.nix
-a----         5/18/2022   5:25 PM           1560 Dockerfile
-a----         5/18/2022   5:25 PM          24268 echidna.png
-a----         5/18/2022   5:25 PM          35184 LICENSE
-a----         5/18/2022   5:25 PM           1129 macos-release.nix
-a----         5/18/2022   5:25 PM           2001 package.yaml
-a----         5/18/2022   5:25 PM          18727 README.md
-a----         5/18/2022   5:25 PM           1270 stack.yaml


PS C:\Users\hamme\Desktop\Smart Contract\slither\echidna> docker build -t echidna .
[+] Building 2.9s (12/21)
 => [internal] load build definition from Dockerfile                                                               0.1s
 => => transferring dockerfile: 1.60kB                                                                             0.0s
 => [internal] load .dockerignore                                                                                  0.1s
 => => transferring context: 66B                                                                                   0.0s
 => [internal] load metadata for docker.io/library/ubuntu:bionic                                                   1.5s
 => [auth] library/ubuntu:pull token for registry-1.docker.io                                                      0.0s
 => [builder 1/7] FROM docker.io/library/ubuntu:bionic@sha256:d21b6ba9e19feffa328cb3864316e6918e30acfd55e285b5d3d  0.0s
 => CACHED [final  2/10] WORKDIR /root                                                                             0.0s
 => [internal] load build context                                                                                  0.4s
 => => transferring context: 656.32kB                                                                              0.3s
 => CACHED [builder 2/7] RUN apt-get update && apt-get -y upgrade && apt-get install -y sudo git cmake curl libgm  0.0s
 => CACHED [builder 3/7] RUN curl -sSL https://get.haskellstack.org/ | sh                                          0.0s
 => CACHED [builder 4/7] COPY . /echidna/                                                                          0.0s
 => CACHED [builder 5/7] WORKDIR /echidna                                                                          0.0s
 => ERROR [builder 6/7] RUN .github/scripts/install-libff.sh                                                       0.8s
------
 > [builder 6/7] RUN .github/scripts/install-libff.sh:
#12 0.634 /bin/sh: 1: .github/scripts/install-libff.sh: not found
------
executor failed running [/bin/sh -c .github/scripts/install-libff.sh]: exit code: 127

**Comment by elopez** [comment]
@FIVIL I see you are building on Windows, can you check if the `.github/scripts/install-libff.sh` file is present in your repository checkout?

**Comment by FIVIL** [comment]
@elopez  sure, as you can see below the files exists, I've also tried building it in Ubuntu using wsl2 but the result was the same and I got the same error.
 `PS C:\Users\hamme\Desktop\Smart Contract\slither\echidna> ls .\.github\scripts\


    Directory: C:\Users\hamme\Desktop\Smart Contract\slither\echidna\.github\scripts


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         5/18/2022   5:25 PM            841 build-macos-release.sh
-a----         5/18/2022   5:25 PM            366 host.sh
-a----         5/18/2022   5:25 PM             83 install-crytic-compile.sh
-a----         5/18/2022   5:25 PM           1173 install-libff.sh
-a----         5/18/2022   5:25 PM            537 install-libsecp256k1.sh
-a----         5/18/2022   5:25 PM            915 install-solc.sh
-a----         5/18/2022   5:25 PM            371 release-binary.sh`

**Comment by elopez** [comment]
@FIVIL Interesting -- If the file exists there, it should be present on the build context used to build the Dockerfile. I wonder if `.dockerignore` is causing an issue here -- it lists `.git`, which might be matching with `.github` somehow?. Would you mind deleting the`.dockerignore` file and trying to build again?

**Comment by elopez** [comment]
@FIVIL yeah, there's certainly something odd going on when copying the context during build -- it looks like not all files get copied to the container layer. Using Docker integration from WSL uses the same instance as the Windows Docker, so that's also something to consider.

Could it be that you are experiencing something like moby/buildkit#1366? If you want to experiment, you can try building from something short like `C:\echidna`, [using `\\?\` paths](https://github.com/moby/buildkit/issues/1366#issuecomment-1047303738) or disabling buildkit (eg. `$Env:DOCKER_BUILDKIT=0`). I'll see if I can try building on Windows when I get some time. It would be nice to narrow the problem down.

**Comment by FIVIL** [comment]
@elopez sorry it took me so long to answer, so I did try building from a shorter path but it failed again.
![image](https://user-images.githubusercontent.com/31272779/169715801-33450c93-e4a3-459f-8a7a-be7521bee9fa.png)

I've also edited the `dockerfile` a bit so that it would only copy the context and not run it, and then I opened the container and I can confirm that all the files are being copied but it is not possible to run the scripts.
![image](https://user-images.githubusercontent.com/31272779/169715882-15b9a95c-6d99-43b0-99b5-91e4e55e0f18.png)
![image](https://user-images.githubusercontent.com/31272779/169715942-c7c0a414-30fc-4cae-8e76-7866f2e19af1.png)
anyway, in the end, I just went ahead and changed the line ending for all the files in the scripts folder from CRLF to LF and now it's building, my guess is that you guys developed everything on Linux, but when someone clones the repo on a windows machine, git automatically converts all the line endings from LF to CRLF which should be done for most files in windows, however, the bash scripts should be eliminated from this process since the `.sh` files aren't being processed on a windows machine and they just get copied to container and execute from there. anyway, the fix is really easy in such a case, I'm gonna go ahead and create a PR with a fix so you guys can check and merge it, I already fixed it in my clone and tried building from my clone on both windows and Linux and it seems fine.
